### ISSUE TITLE
Flutter app with Swift support fails to build with shared_preferences plugin

### ISSUE THREAD

**Issue by kibantony** [flutter/flutter#15175]
## Steps to Reproduce

Trying to create my first flutter app, I created a new project in Android Studio, but when I added the shared_preferences plugin to my pubspec.yaml, the iOS build failed. I managed to boil down a 100% report case to the following command line steps:

flutter create -i swift hello
cd hello
[edit pubspec.yaml to add shared_preferences: ^0.3.3]

(I suspect the issue may occur with any obj-c based plugin, not just shared_preferences, fwiw.)

## Logs

```
$ flutter run
Running "flutter packages get" in hello...            1.2s
Launching lib/main.dart on iPhone X in debug mode...
Running pod install...                                1.2s
Running Xcode clean...                                1.1s
Starting Xcode build...                                   
 ├─Assembling Flutter resources...             4.1s
 └─Compiling, linking and signing...           5.4s
Xcode build done                                     12.4s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
    
Xcode's output:
↳
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    The use of Swift 3 @objc inference in Swift 4 mode is deprecated. Please address deprecated @objc inference warnings, test your code with “Use of deprecated Swift 3 @objc inference” logging enabled, and then disable inference by changing the "Swift 3 @objc Inference" build setting to "Default" for the "Runner" target.
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    /Users/tony/afero/hello/ios/Runner/GeneratedPluginRegistrant.m:6:9: fatal error: 'shared_preferences/SharedPreferencesPlugin.h' file not found
    #import <shared_preferences/SharedPreferencesPlugin.h>
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
Could not build the application for the simulator.
Error launching application on iPhone X.
```

## Flutter Doctor

[✓] Flutter (Channel beta, v0.1.5, on Mac OS X 10.12.6 16G1212, locale en-US)
    • Flutter version 0.1.5 at /Users/tony/afero/flutter
    • Framework revision 3ea4d06340 (11 days ago), 2018-02-22 11:12:39 -0800
    • Engine revision ead227f118
    • Dart version 2.0.0-dev.28.0.flutter-0b4f01f759

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.2)
    • Android SDK at /Users/tony/Library/Android/sdk
    • Android NDK at /Users/tony/Library/Android/sdk/ndk-bundle
    • Platform android-27, build-tools 26.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.4.0

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] Connected devices (1 available)
    • iPhone X • 86F5D909-F2B9-46E3-B177-039CEE3C668A • ios • iOS 11.2 (simulator)

• No issues found!

**Comment by kibantony** [comment]
Nevermind, looks like a duplicate of #15099

**Comment by mravn-google** [comment]
Closed as duplicate of #15099
